src/main/kotlin/pt/up/fe/ni/website/backend/model/Account.kt

```diff
@@ -56,6 +58,24 @@ class Account(
     @OneToMany(cascade = [CascadeType.ALL], fetch = FetchType.EAGER)
     val websites: List<@Valid CustomWebsite>,
 
+    @JoinColumn
+    @ManyToMany(cascade = [CascadeType.ALL], fetch = FetchType.EAGER)
+    val roles: List<@Valid Role> = emptyList(),
+
     @Id @GeneratedValue
     val id: Long? = null
-)
+) {
+    fun getEffectivePermissionsForActivity(activity: Activity) {
```

Comment:
Since we have this in the model, it might be a good idea to unit test it. What do you think?